calico_containers/pycalico/util.py

```diff
@@ -104,211 +108,450 @@ def get_hostname():
         return socket.gethostname()
 
 
-def validate_port_str(port_str):
+def return_bool(func, *args, **kwargs):
     """
-    Checks whether the command line word specifying a set of ports is valid.
+    Simple function to catch exceptions and return a Bool
+
+    :param func: The function being run
+    :param *args: All positional args being passed to func
+    :param **kwargs: All key-value args being passed to func
+    :return: True if function succeeds, False if exception is raised
+    :rtype: bool
     """
-    return validate_ports(port_str.split(","))
+    try:
+        func(*args, **kwargs)
+    except Exception:  # pylint: disable=broad-except
```

Comment:
I'd like us to use a custom Exception class for this (VerifyError seems like the obvious choice).  Motivation being that it propagates unforeseen Exceptions up the stack; clearly distinguishing verification failures (user error) from coding errors.  It also reduces risk that someone writing against this module misses an exception.  We do lose a little expressiveness (currently TypeError vs ValueError), but I think for user input validation the thing that really matters is the message.